ampd/src/sui/verifier.rs

```diff
@@ -0,0 +1,194 @@
+use move_core_types::language_storage::StructTag;
+use serde::Deserialize;
+use sui_json_rpc_types::{SuiEvent, SuiTransactionBlockResponse};
+use sui_types::base_types::{ObjectID, SuiAddress};
+
+use crate::handlers::sui_verify_msg::Message;
+use crate::types::Hash;
+
+const CONTRACT_CALL_EVENT: &str = "gateway::ContractCall";
+
+// TODO: update after Sui gateway event finalization
+#[derive(Deserialize)]
+struct ContractCall {
+    pub source_id: SuiAddress,
+    pub destination_chain: String,
+    pub destination_address: String,
+    pub payload_hash: Hash,
+}
+
+// event type is in the form of: <gateway_address>::gateway::ContractCall
+fn call_contract_type(gateway_address: &ObjectID) -> StructTag {
+    format!("{}::{}", gateway_address, CONTRACT_CALL_EVENT)
+        .as_str()
+        .parse()
+        .expect("failed to parse struct tag")
+}
+
+impl PartialEq<&Message> for &SuiEvent {
+    fn eq(&self, msg: &&Message) -> bool {
+        if self.type_ != call_contract_type(&self.package_id) {
+            return false;
+        }
+
+        serde_json::from_value::<ContractCall>(self.parsed_json.clone()).unwrap();
```

Comment:
no `unwrap` out side of tests

Reply:
maybe this is a leftover from testing? it doesn't do anything

Reply:
Also, why this extra deserialization when the same thing is being matched below?

Reply:
leftover from testing 😓, this line is unnecessary